From 26afc621069c31a23922dce3bfbc1c733ba72f5a Mon Sep 17 00:00:00 2001
From: Naveen Gangam <ngangam@cloudera.com>
Date: Tue, 13 Feb 2018 10:55:49 -0500
Subject: [PATCH 1345/1363] Revert "CDH-57894 HIVE-17218: Canonical-ize
 hostnames for Hive metastore, and HS2 servers."

This reverts commit b2ffecc5f6290d569cc354a5e72e0674255e6a0c.

Change-Id: Ic854dd2ad15556a9b70ead00ff765df47f9bd9e2
(cherry picked from commit df4f515720b3314963f321f0339fa9befe546ce2)
---
 .../java/org/apache/hive/jdbc/HiveConnection.java  |    3 +--
 .../hadoop/hive/metastore/HiveMetaStoreClient.java |   10 +---------
 .../hadoop/hive/thrift/HadoopThriftAuthBridge.java |   16 ----------------
 3 files changed, 2 insertions(+), 27 deletions(-)

diff --git a/jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java b/jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java
index e2cf752..1d5d785 100644
--- a/jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java
+++ b/jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java
@@ -19,7 +19,6 @@
 package org.apache.hive.jdbc;
 
 import org.apache.hadoop.hive.common.auth.HiveAuthUtils;
-import org.apache.hadoop.hive.shims.ShimLoader;
 import org.apache.hive.jdbc.Utils.JdbcConnectionParams;
 import org.apache.hive.service.auth.HiveAuthFactory;
 import org.apache.hive.service.auth.KerberosSaslHelper;
@@ -130,7 +129,7 @@ public HiveConnection(String uri, Properties info) throws SQLException {
     // sess_var_list -> sessConfMap
     // hive_conf_list -> hiveConfMap
     // hive_var_list -> hiveVarMap
-    host = ShimLoader.getHadoopThriftAuthBridge().getCanonicalHostName(connParams.getHost());
+    host = connParams.getHost();
     port = connParams.getPort();
     sessConfMap = connParams.getSessionVars();
     hiveConfMap = connParams.getHiveConfs();
diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java b/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java
index c77d603..7cdb082 100644
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java
@@ -228,15 +228,7 @@ public HiveMetaStoreClient(HiveConf conf, HiveMetaHookLoader hookLoader)
             throw new IllegalArgumentException("URI: " + s
                 + " does not have a scheme");
           }
-          metastoreUris[i++] = new URI(
-              tmpUri.getScheme(),
-              tmpUri.getUserInfo(),
-              ShimLoader.getHadoopThriftAuthBridge().getCanonicalHostName(tmpUri.getHost()),
-              tmpUri.getPort(),
-              tmpUri.getPath(),
-              tmpUri.getQuery(),
-              tmpUri.getFragment()
-          );
+          metastoreUris[i++] = tmpUri;
 
         }
       } catch (IllegalArgumentException e) {
diff --git a/shims/common/src/main/java/org/apache/hadoop/hive/thrift/HadoopThriftAuthBridge.java b/shims/common/src/main/java/org/apache/hadoop/hive/thrift/HadoopThriftAuthBridge.java
index 4956087..2e80171 100644
--- a/shims/common/src/main/java/org/apache/hadoop/hive/thrift/HadoopThriftAuthBridge.java
+++ b/shims/common/src/main/java/org/apache/hadoop/hive/thrift/HadoopThriftAuthBridge.java
@@ -22,7 +22,6 @@
 import java.io.IOException;
 import java.net.InetAddress;
 import java.net.Socket;
-import java.net.UnknownHostException;
 import java.security.PrivilegedAction;
 import java.security.PrivilegedExceptionAction;
 import java.util.Locale;
@@ -116,21 +115,6 @@ public String getServerPrincipal(String principalConfig, String host)
     return serverPrincipal;
   }
 
-  /**
-   * Method to get canonical-ized hostname, given a hostname (possibly a CNAME).
-   * This should allow for service-principals to use simplified CNAMEs.
-   * @param hostName The hostname to be canonical-ized.
-   * @return Given a CNAME, the canonical-ized hostname is returned. If not found, the original hostname is returned.
-   */
-  public String getCanonicalHostName(String hostName) {
-    try {
-      return InetAddress.getByName(hostName).getCanonicalHostName();
-    }
-    catch(UnknownHostException exception) {
-      LOG.warn("Could not retrieve canonical hostname for " + hostName, exception);
-      return hostName;
-    }
-  }
 
   public UserGroupInformation getCurrentUGIWithConf(String authMethod)
       throws IOException {
-- 
1.7.9.5

